packages/apidom-ls/src/services/validation/linter-functions.ts

```diff
@@ -1009,9 +1011,35 @@ export const standardLinterfunctions: FunctionItem[] = [
     functionName: 'apilintOpenAPIPathTemplateValid',
     function: (element: Element) => {
       if (isStringElement(element)) {
-        return true;
+        const pathItemElement = (element.parent as MemberElement).value as ObjectElement;
+
+        if (pathItemElement.length === 0) {
+          return true;
+        }
+
+        const pathTemplateResolveParams: { [key: string]: string } = {};
+
+        pathItemElement.forEach((operation) => {
+          const parameters = (operation as OperationElement).get('parameters');
+          if (isArrayElement(parameters)) {
+            parameters.forEach((parameter) => {
+              if (isParameterElement(parameter)) {
```

Comment:
We cannot use semantic predicates here. This linting rule is not specific to any particular OpenAPI version, but rather is general to all OpenAPI versions (2.0, 3.0, 3.1). `isParameterElement` is specific to OpenAPI 3.0.x namespace and will not detect `ParamterElement` from OpenAPI 2.0.

We need to polymorphic check which will look like this:

```js
if (parameter.element === 'parameter') {
```